app/src/main/java/org/mozilla/focus/utils/StorageUtils.java

```diff
@@ -27,21 +50,14 @@
      * @return a directory on removable storage to save files. return null if user's preference is off.
      * @throws NoRemovableStorageException if user's preference is on, but there is no removable storage to use.
      */
-    public static File getTargetDirOnRemovableStorage(@NonNull Context ctx, String type)
+    public static File getTargetDirOnRemovableStorageForDownloads(@NonNull Context ctx, String type)
             throws NoRemovableStorageException {
 
         if (!Settings.getInstance(ctx).shouldSaveToRemovableStorage()) {
             return null;
         }
 
-        final File media = getFirstRemovableMedia(ctx);
-        if (media == null) {
-            throw new NoRemovableStorageException("No removable media to use");
-        }
-
-        if (!Environment.MEDIA_MOUNTED.equals(Environment.getExternalStorageState(media))) {
-            throw new NoRemovableStorageException("No mounted-removable media to use");
-        }
+        final File media = testAndGetRemovableStorage(ctx);
 
         final File dir = new File(media, DOWNLOAD_DIR);
```

Comment:
We can refactor this method by removing this line, then rename to 
```diff
- public static File getTargetDirOnRemovableStorage(@NonNull Context ctx, String type)
+ public static File getAppMediaDirOnRemovableStorage(@NonNull Context ctx)
```

Then add this two functions
```java
public static File getTargetDirOnRemovableStorageForDownloads(@NonNull Context ctx, String mimeType) throws NoRemovableStorageException {
    File dir = getAppMediaDirOnRemovableStorage(ctx);
    if (dir == null) {
        return null;
    } else {
        return (MimeUtils.isImage(type)) ?  new File(dir, IMAGE_DIR) : new File(dir, OTHER_DIR);
    }
}

public static File getDirForSaveFile(@NonNull Context ctx, String folderName) {
    File dir = getAppMediaDirOnRemovableStorage(ctx);
    if (dir == null OR NoRemovableStorageException) {
        // user don't want to save to external dir, fallback to internal storage
        return new File(Envrionment....., folderName);
    } else {
        return new File(dir, folderName);
    }
}
```

Reply:
Per our discussion, this is reasonable.